# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build cairo-build harfbuzz-old-build"

PKG_CONFIG_PATH=/x11/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.63; am-1.11; lt-2.2.6)
libtoolize -c -f; aclocal -I m4; autoconf; autoheader -f; automake -a -c
rm -rf pango/opentype
PKG_CONFIG_PATH=/x11/lib/pkgconfig ./configure --with-system-harfbuzz --prefix=/x11

# build
make

# install
install -d $DESTDIR/x11/bin
install pango/.libs/pango-querymodules $DESTDIR/x11/bin
install pango-view/.libs/pango-view $DESTDIR/x11/bin
install -d $DESTDIR/x11/etc/pango
install -d $DESTDIR/x11/include/pango-1.0/pango
install -m 644 pango/pango.h $DESTDIR/x11/include/pango-1.0/pango
install -m 644 pango/pango-attributes.h $DESTDIR/x11/include/pango-1.0/pango
install -m 644 pango/pango-bidi-type.h $DESTDIR/x11/include/pango-1.0/pango
install -m 644 pango/pango-break.h $DESTDIR/x11/include/pango-1.0/pango
install -m 644 pango/pango-context.h $DESTDIR/x11/include/pango-1.0/pango
install -m 644 pango/pango-coverage.h $DESTDIR/x11/include/pango-1.0/pango
install -m 644 pango/pango-engine.h $DESTDIR/x11/include/pango-1.0/pango
install -m 644 pango/pango-enum-types.h $DESTDIR/x11/include/pango-1.0/pango
install -m 644 pango/pango-features.h $DESTDIR/x11/include/pango-1.0/pango
install -m 644 pango/pango-font.h $DESTDIR/x11/include/pango-1.0/pango
install -m 644 pango/pango-fontmap.h $DESTDIR/x11/include/pango-1.0/pango
install -m 644 pango/pango-fontset.h $DESTDIR/x11/include/pango-1.0/pango
install -m 644 pango/pango-glyph-item.h $DESTDIR/x11/include/pango-1.0/pango
install -m 644 pango/pango-glyph.h $DESTDIR/x11/include/pango-1.0/pango
install -m 644 pango/pango-gravity.h $DESTDIR/x11/include/pango-1.0/pango
install -m 644 pango/pango-item.h $DESTDIR/x11/include/pango-1.0/pango
install -m 644 pango/pango-language.h $DESTDIR/x11/include/pango-1.0/pango
install -m 644 pango/pango-layout.h $DESTDIR/x11/include/pango-1.0/pango
install -m 644 pango/pango-matrix.h $DESTDIR/x11/include/pango-1.0/pango
install -m 644 pango/pango-renderer.h $DESTDIR/x11/include/pango-1.0/pango
install -m 644 pango/pango-script.h $DESTDIR/x11/include/pango-1.0/pango
install -m 644 pango/pango-tabs.h $DESTDIR/x11/include/pango-1.0/pango
install -m 644 pango/pango-types.h $DESTDIR/x11/include/pango-1.0/pango
install -m 644 pango/pango-utils.h $DESTDIR/x11/include/pango-1.0/pango
install -m 644 pango/pangocairo.h $DESTDIR/x11/include/pango-1.0/pango
install -m 644 pango/pangofc-decoder.h $DESTDIR/x11/include/pango-1.0/pango
install -m 644 pango/pangofc-font.h $DESTDIR/x11/include/pango-1.0/pango
install -m 644 pango/pangofc-fontmap.h $DESTDIR/x11/include/pango-1.0/pango
install -m 644 pango/pangoft2.h $DESTDIR/x11/include/pango-1.0/pango
install -d $DESTDIR/x11/lib
install pango/.libs/libpango-1.0.so.0.* $DESTDIR/x11/lib/libpango-1.0.so.0
ln -sf libpango-1.0.so.0 $DESTDIR/x11/lib/libpango-1.0.so
install pango/.libs/libpangocairo-1.0.so.0.* $DESTDIR/x11/lib/libpangocairo-1.0.so.0
ln -sf libpangocairo-1.0.so.0 $DESTDIR/x11/lib/libpangocairo-1.0.so
install pango/.libs/libpangoft2-1.0.so.0.* $DESTDIR/x11/lib/libpangoft2-1.0.so.0
ln -sf libpangoft2-1.0.so.0 $DESTDIR/x11/lib/libpangoft2-1.0.so
install -d $DESTDIR/x11/lib/pango/1.6.0/modules
install modules/basic/.libs/pango-basic-fc.so $DESTDIR/x11/lib/pango/1.6.0/modules
install -d $DESTDIR/x11/lib/pkgconfig
install -m 644 pango.pc $DESTDIR/x11/lib/pkgconfig
install -m 644 pangocairo.pc $DESTDIR/x11/lib/pkgconfig
install -m 644 pangoft2.pc $DESTDIR/x11/lib/pkgconfig
LD_LIBRARY_PATH=$DESTDIR/x11/lib $DESTDIR/x11/bin/pango-querymodules > $DESTDIR/x11/etc/pango/pango.modules

# build.pc
install -d $DESTDIR/x11/share/pkgconfig
cat > $DESTDIR/x11/share/pkgconfig/pango-build.pc << EOF
Name: pango
Version: 1.24.5
Description: A library for layout and rendering of text
Requires: $REQUIRES

devel=\\
/x11/include/pango-1.0/pango/pango.h \\
/x11/include/pango-1.0/pango/pango-attributes.h \\
/x11/include/pango-1.0/pango/pango-bidi-type.h \\
/x11/include/pango-1.0/pango/pango-break.h \\
/x11/include/pango-1.0/pango/pango-context.h \\
/x11/include/pango-1.0/pango/pango-coverage.h \\
/x11/include/pango-1.0/pango/pango-engine.h \\
/x11/include/pango-1.0/pango/pango-enum-types.h \\
/x11/include/pango-1.0/pango/pango-features.h \\
/x11/include/pango-1.0/pango/pango-font.h \\
/x11/include/pango-1.0/pango/pango-fontmap.h \\
/x11/include/pango-1.0/pango/pango-fontset.h \\
/x11/include/pango-1.0/pango/pango-glyph-item.h \\
/x11/include/pango-1.0/pango/pango-glyph.h \\
/x11/include/pango-1.0/pango/pango-gravity.h \\
/x11/include/pango-1.0/pango/pango-item.h \\
/x11/include/pango-1.0/pango/pango-language.h \\
/x11/include/pango-1.0/pango/pango-layout.h \\
/x11/include/pango-1.0/pango/pango-matrix.h \\
/x11/include/pango-1.0/pango/pango-renderer.h \\
/x11/include/pango-1.0/pango/pango-script.h \\
/x11/include/pango-1.0/pango/pango-tabs.h \\
/x11/include/pango-1.0/pango/pango-types.h \\
/x11/include/pango-1.0/pango/pango-utils.h \\
/x11/include/pango-1.0/pango/pangocairo.h \\
/x11/include/pango-1.0/pango/pangofc-decoder.h \\
/x11/include/pango-1.0/pango/pangofc-font.h \\
/x11/include/pango-1.0/pango/pangofc-fontmap.h \\
/x11/include/pango-1.0/pango/pangoft2.h \\
/x11/lib/libpango-1.0.so \\
/x11/lib/libpangocairo-1.0.so \\
/x11/lib/libpangoft2-1.0.so \\
/x11/lib/pkgconfig/pango.pc \\
/x11/lib/pkgconfig/pangocairo.pc \\
/x11/lib/pkgconfig/pangoft2.pc

exec=\\
/x11/bin/pango-querymodules \\
/x11/bin/pango-view \\
/x11/etc/pango/pango.modules \\
/x11/lib/libpango-1.0.so.0 \\
/x11/lib/libpangocairo-1.0.so.0 \\
/x11/lib/libpangoft2-1.0.so.0 \\
/x11/lib/pango/1.6.0/modules/pango-basic-fc.so
EOF

exit
--- pango-1.24.5.orig/configure.in
+++ pango-1.24.5/configure.in
@@ -42,6 +42,7 @@
         [http://bugzilla.gnome.org/enter_bug.cgi?product=pango])
 
 AC_CONFIG_SRCDIR([pango.pc.in])
+AC_CONFIG_MACRO_DIR([m4])
 
 dnl -Wno-portability tells automake to stop complaining about the
 dnl usage of GNU Make specific features.
@@ -199,6 +200,10 @@
                               [disable all source autogeneration rules])],,
               [enable_rebuilds=yes])
 
+AC_ARG_WITH(system-harfbuzz,
+   AC_HELP_STRING([--with-system-harfbuzz],
+                  [use system harfbuzz library @<:@default=no@:>@]))
+
 AM_CONDITIONAL(CROSS_COMPILING, [test $cross_compiling = yes])
 
 # define a MAINT-like variable REBUILD which is set if Perl
@@ -251,7 +256,17 @@
   # By freetype, we really always mean freetype+fontconfig
   FREETYPE_CFLAGS="$FREETYPE_CFLAGS $FONTCONFIG_CFLAGS"
   FREETYPE_LIBS="$FREETYPE_LIBS $FONTCONFIG_LIBS"
+
+  if test "x$with_system_harfbuzz" = "xyes"; then
+    PKG_CHECK_MODULES(OPENTYPE, harfbuzz-1)
+  else
+    OPENTYPE_LIBS="opentype/libharfbuzz-1.la"
+    OPENTYPE_DEPS="$OPENTYPE_LIBS"
+    OPENTYPE_SUBDIR=opentype
+  fi
 fi
+AC_SUBST(OPENTYPE_DEPS)
+AC_SUBST(OPENTYPE_SUBDIR)
 AM_CONDITIONAL(HAVE_FREETYPE, $have_freetype)
 
 #
@@ -964,7 +979,6 @@
 Makefile 
 pango/Makefile
 pango/mini-fribidi/Makefile
-pango/opentype/Makefile
 pango/pango.rc
 pango/pangoft2.rc
 pango/pangowin32.rc
@@ -999,6 +1013,10 @@
 pango-zip.sh
 ])
 
+if test "x$with_system_harfbuzz" != "xyes"; then
+  AC_CONFIG_FILES([pango/opentype/Makefile])
+fi
+
 AC_CONFIG_FILES([tests/runtests.sh],
 	        [chmod +x tests/runtests.sh])
 
--- pango-1.24.5.orig/m4/gtk-doc.m4
+++ pango-1.24.5/m4/gtk-doc.m4
@@ -0,0 +1,39 @@
+dnl -*- mode: autoconf -*-
+
+# serial 1
+
+dnl Usage:
+dnl   GTK_DOC_CHECK([minimum-gtk-doc-version])
+AC_DEFUN([GTK_DOC_CHECK],
+[
+  AC_BEFORE([AC_PROG_LIBTOOL],[$0])dnl setup libtool first
+  AC_BEFORE([AM_PROG_LIBTOOL],[$0])dnl setup libtool first
+  dnl for overriding the documentation installation directory
+  AC_ARG_WITH([html-dir],
+    AS_HELP_STRING([--with-html-dir=PATH], [path to installed docs]),,
+    [with_html_dir='${datadir}/gtk-doc/html'])
+  HTML_DIR="$with_html_dir"
+  AC_SUBST([HTML_DIR])
+
+  dnl enable/disable documentation building
+  AC_ARG_ENABLE([gtk-doc],
+    AS_HELP_STRING([--enable-gtk-doc],
+                   [use gtk-doc to build documentation [[default=no]]]),,
+    [enable_gtk_doc=no])
+
+  if test x$enable_gtk_doc = xyes; then
+    ifelse([$1],[],
+      [PKG_CHECK_EXISTS([gtk-doc],,
+                        AC_MSG_ERROR([gtk-doc not installed and --enable-gtk-doc requested]))],
+      [PKG_CHECK_EXISTS([gtk-doc >= $1],,
+                        AC_MSG_ERROR([You need to have gtk-doc >= $1 installed to build gtk-doc]))])
+  fi
+
+  AC_MSG_CHECKING([whether to build gtk-doc documentation])
+  AC_MSG_RESULT($enable_gtk_doc)
+
+  AC_PATH_PROGS(GTKDOC_CHECK,gtkdoc-check,)
+
+  AM_CONDITIONAL([ENABLE_GTK_DOC], [test x$enable_gtk_doc = xyes])
+  AM_CONDITIONAL([GTK_DOC_USE_LIBTOOL], [test -n "$LIBTOOL"])
+])
--- pango-1.24.5.orig/pango/Makefile.am
+++ pango-1.24.5/pango/Makefile.am
@@ -4,10 +4,6 @@
 # which are more like other make's VPATH.
 GPATH = $(srcdir)
 
-if HAVE_FREETYPE
-OPENTYPE_SUBDIR=opentype
-endif
-
 EXTRA_DIST = 
 BUILT_GIRSOURCES =
 
@@ -15,7 +11,7 @@
 
 SUBDIRS = $(OPENTYPE_SUBDIR) $(FRIBIDI_SUBDIR)
 
-DIST_SUBDIRS = mini-fribidi opentype
+DIST_SUBDIRS = mini-fribidi
 
 INCLUDES =						\
 	-DG_LOG_DOMAIN=\"Pango\"			\
@@ -251,16 +247,17 @@
 	pangofc-decoder.c	\
 	pangoft2.c
 
+libpangoft2_1_0_la_CFLAGS = $(OPENTYPE_CFLAGS)
 libpangoft2_1_0_la_LDFLAGS = $(LIBRARY_LIBTOOL_OPTIONS)
 libpangoft2_1_0_la_LIBADD =			\
-	opentype/libharfbuzz-1.la		\
+	$(OPENTYPE_LIBS)			\
 	libpango-$(PANGO_API_VERSION).la	\
 	$(INCLUDED_FC_MODULES)			\
 	$(GLIB_LIBS)				\
 	$(FREETYPE_LIBS)			\
 	$(libm)
 libpangoft2_1_0_la_DEPENDENCIES =		\
-	opentype/libharfbuzz-1.la		\
+	$(OPENTYPE_DEPS)			\
 	libpango-$(PANGO_API_VERSION).la	\
 	$(INCLUDED_FC_MODULES)
 libpangoft2_1_0_la_SOURCES =   	\
